packages/ic-asset-certification/src/asset_config.rs

```diff
@@ -324,6 +403,97 @@ pub enum AssetRedirectKind {
     Temporary,
 }
 
+/// The encoding of an asset.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum AssetEncoding {
+    /// The asset is encoded with the Brotli algorithm.
+    Brotli,
+
+    /// The asset is encoded with the Zstd algorithm.
+    Zstd,
+
+    /// The asset is encoded with the Gzip algorithm.
+    Gzip,
+
+    /// The asset is encoded with the Deflate algorithm.
+    Deflate,
+}
+
+impl AssetEncoding {
+    /// Returns the default encoding and file extension for the encoding.
+    /// The default encoding is the encoding that is used when the client
+    /// does not specify an encoding in the `Accept-Encoding` header.
+    ///
+    /// The default encoding and file extension are:
+    /// - [Brotli](AssetEncoding::Brotli): `br`
+    /// - [Zstd](AssetEncoding::Zstd): `zst`
+    /// - [Gzip](AssetEncoding::Gzip): `gz`
+    /// - [Deflate](AssetEncoding::Deflate): `zz`
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use ic_asset_certification::AssetEncoding;
+    ///
+    /// let (encoding, extension) = AssetEncoding::Brotli.default();
+    /// assert_eq!(encoding, AssetEncoding::Brotli);
+    /// assert_eq!(extension, "br");
+    ///
+    /// let (encoding, extension) = AssetEncoding::Zstd.default();
+    /// assert_eq!(encoding, AssetEncoding::Zstd);
+    /// assert_eq!(extension, "zst");
+    ///
+    /// let (encoding, extension) = AssetEncoding::Gzip.default();
+    /// assert_eq!(encoding, AssetEncoding::Gzip);
+    /// assert_eq!(extension, "gz");
+    ///
+    /// let (encoding, extension) = AssetEncoding::Deflate.default();
+    /// assert_eq!(encoding, AssetEncoding::Deflate);
+    /// assert_eq!(extension, "zz");
+    /// ```
+    pub fn default(self) -> (AssetEncoding, String) {
```

Comment:
nit: I like the new approach, but maybe rename `default()` to something else, say `default_extension()` or `default_config()`, as `default()` is "reserved" by `Default`-trait.

Reply:
Done, thanks!